### PR TITLE
move warning flags to target compile options

### PR DESCRIPTION
**Pull request by mathiaswagner**
This applies the warning flags only to the targets defined in QUDA's CMakeLists and should help with warning from external dependencies as well as fixing #1305.

Note that this might trigger some warning that have previously gone unnoticed.

### CODE REVIEW

**Comment by mathiaswagner** [comment]
@Jenkins test this please

**Review by maddyscientist — APPROVED** [review]
Thanks @mathiaswagner.

I have fixed the warnings incurred with this PR, so we should be good to go once Jenkins completes.

File: include/gauge_field_order.h [hunk 1]
```diff
@@ -1552,14 +1549,10 @@ namespace quda {
           for (int j = 0; j < N; j++) copy(tmp[i * N + j], reinterpret_cast<Float *>(&vecTmp)[j]);
         }
 
-        real phase = 0.;
```
Comment (mathiaswagner) [inline include/gauge_field_order.h:1555]: just wondering as I had looked at that warning before: For `!hasPhase` we had `phase=0.` in the code and now it seems that has gone. I have not looked at it in detail but did not want to forget about it.

Comment (maddyscientist) [reply include/gauge_field_order.h:1555]: Yeah, it doesn’t affect anything.  For the non-phased reconstruct types, it doesn’t use the phase value at all so it doesn’t matter what this is set to.  I checked that the ctest staggered and Wilson tests all still pass with a full reconstruct build.

File: include/kernels/dslash_wilson2.cuh [hunk 2]
```diff
@@ -0,0 +1,249 @@
+#pragma once
```
Comment (mathiaswagner) [inline include/kernels/dslash_wilson2.cuh]: Why do we need this file?

Comment (maddyscientist) [reply include/kernels/dslash_wilson2.cuh]: Accidental addition.  Can fix when I’m back 💻

**Review by mathiaswagner — COMMENTED** [review]
Looks good now

File: include/kernels/dslash_wilson2.cuh (hunk 2 above)
Comment (mathiaswagner) [reply include/kernels/dslash_wilson2.cuh]: I just removed it.

**Pull request merged**